openmc/mgxs/mgxs.py

```diff
@@ -5011,13 +5028,1153 @@ class NuFissionMatrixXS(MatrixMGXS):
+                 by_nuclide=False, name='', num_polar=1, num_azimuthal=1):
+        super(ConsistentScatterMatrixXS, self).__init__(
+            domain, domain_type, groups, by_nuclide,
+            name, num_polar, num_azimuthal)
+
+        self._hdf5_key = 'consistent scatter matrix'
+
+        # Initialize each MGXS used by the convolution
+        self._mgxs = [ScatterXS(), ScatterProbabilityMatrix()]
+
+        # Assign parameters to each MGXS in the convlution
+        for mgxs in self.mgxs:
+            mgxs.name = name
+            mgxs.by_nuclide = by_nuclide
+
+            if domain_type is not None:
+                mgxs.domain_type = domain_type
+            if domain is not None:
+                mgxs.domain = domain
+            if groups is not None:
+                mgxs.energy_groups = groups
+            mgxs.num_polar = num_polar
+            mgxs.num_azimuthal = num_azimuthal
+
+    @property
+    def scores(self):
+        scores = super(ConsistentScatterMatrixXS, self).scores
+
+        # Add scores for transport correction
+        if self.correction == 'P0' and self.legendre_order == 0:
+            scores += ['{}-1'.format(self.rxn_type)]
+
+        return scores
+
+    @property
+    def filters(self):
+        filters = super(ConsistentScatterMatrixXS, self).filters
+
+        # Add filters for transport correction
+        if self.correction == 'P0' and self.legendre_order == 0:
+            group_edges = self.energy_groups.group_edges
+            energyout = openmc.EnergyoutFilter(group_edges)
+            filters += self._add_angle_filters([[energyout]])
+
+        return filters
+
+    @property
+    def tally_keys(self):
+        tally_keys = super(ConsistentScatterMatrixXS, self).tally_keys
+
+        # Add key for transport correction tally
+        if self.correction == 'P0' and self.legendre_order == 0:
+            tally_keys += ['scatter-1']
+
+        return tally_keys
+
+    @property
+    def tallies(self):
+
+        if self._tallies is None:
+            self._tallies = super(ConsistentScatterMatrixXS, self).tallies
+
+            # Add in the transport correction tally
+            if self.correction == 'P0' and self.legendre_order == 0:
+                tally_key = 'scatter-1'
+
+                # Create a domain Filter object
+                filter_type = _DOMAIN_TO_FILTER[self.domain_type]
+                if self.domain_type == 'mesh':
+                    domain_filter = filter_type(self.domain)
+                else:
+                    domain_filter = filter_type(self.domain.id)
+
+                # Create each Tally needed to compute the multi group cross section
+                self._tallies['scatter-1'] = openmc.Tally(name=self.name)
+                self._tallies['scatter-1'].estimator = 'analog'
+                self._tallies['scatter-1'].scores = [self.scores[-1]]
+                self._tallies['scatter-1'].filters = [domain_filter, *self.filters[-1]]
```

Comment:
This is the line that is causing Travis to fail. If this is really what you intended, this type of argument unpacking didn't become acceptable until Python 3.5.